cdap-docs/developer-manual/source/metadata/programmatic-metadata.rst

```diff
@@ -0,0 +1,91 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2018 Cask Data, Inc.
+
+.. _programmatic-metadata:
+
+==================================
+Metadata from Programs and Plugins
+==================================
+
+.. highlight:: java
+
+Metadata can be accessed from *Programs* and *Plugins*. Currently, reading metadata from a program or plugin is not
+supported while it is running in *cloud* but metadata can be added, updated or deleted.
+
+.. _metadata-programs:
+
+Programs
+========
+
+Metadata can be accessed from *MapReduce*, *Spark*, *Workers* and *Service* through
+`MetadataReader <../../reference-manual/javadocs/co/cask/cdap/api/metadata/MetadataReader.html>`__ exposed to
+individual program context available in the ``initialize`` method.
+
+For example metadata can be read in *MapReduce* programs in the following way::
```

Comment:
The following example shows how you can retrieve the metadata of an entity in a *MapReduce* program::

Reply:
Done.